### PR TITLE
Add Hover State to InfoListItems with onClick Event

### LINKED ISSUE #36: InfoListItem needs a hover state when an onClick is present
#### Describe the bug
No hover state on InfoListItem 

#### What is the expected behavior?
Should show a darker background when hovered and onClick event is defined
		
#### Anything else to add?
This is related to a recent change in the theme, which removed the default hover state. We want to apply this style back, but only when onClick is defined

### PR DESCRIPTION
**Pull request by emclaug2**
Fixes #36 

Changes proposed in this Pull Request:
- Remove most inline styles so that they don't conflict with potential style overrides.
- Add hover state when a click event is present.

### CODE REVIEW

File: components/src/core/InfoListItem/InfoListItem.styles.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+export const useStyles = makeStyles<Theme, InfoListItemProps>((theme: Theme) =>
+    createStyles({
+        root: {
+            backgroundColor: (props): string => props.backgroundColor || 'transparent',
+            cursor: (props): string => (props.onClick ? 'pointer' : 'default'),
+            height: (props): number => (props.dense ? 52 : 72),
+            '&:hover': {
+                backgroundColor: (props): string => (props.onClick ? 'rgba(0, 0, 0, 0.08)' : ''),
```
Comment (joebochill) [inline components/src/core/InfoListItem/InfoListItem.styles.tsx]: how's this work for different themes?

File: components/src/core/InfoListItem/InfoListItem.test.tsx [hunk 2]
```diff
@@ -41,10 +39,11 @@ describe('InfoListItem', () => {
         wrapper = shallow(<InfoListItem hidePadding title="Test" />);
         expect(wrapper.find(PersonIcon).length).toEqual(0);
     });
-
+/* No longer using in-line props to modify styling.
```
Comment (joebochill) [inline components/src/core/InfoListItem/InfoListItem.test.tsx]: Can we find another way to test this, or add another test of some sort?

File: components/src/core/InfoListItem/InfoListItem.tsx [hunk 3]
```diff
@@ -98,58 +52,34 @@ export type InfoListItemProps = {
...
                 <ListItemAvatar>
-                    <Avatar
-                        style={{
-                            backgroundColor: statusColor || Colors.black[500],
-                            color: getIconColor(),
-                        }}
-                    >
-                        {icon}
-                    </Avatar>
+                    <Avatar className={defaultClasses.avatar}>{icon}</Avatar>
```
Comment (joebochill) [inline components/src/core/InfoListItem/InfoListItem.tsx]: if we're moving these into classes, can we exposes entries in the classes prop for users?

File: components/src/core/InfoListItem/InfoListItem.tsx [hunk 4]
```diff
@@ -98,58 +52,34 @@ export type InfoListItemProps = {
...
                 </ListItemAvatar>
             );
         } else if (icon) {
-            return <ListItemIcon style={{ color: getIconColor() }}>{icon}</ListItemIcon>;
+            return <ListItemIcon className={defaultClasses.icon}>{icon}</ListItemIcon>;
```
Comment (joebochill) [inline components/src/core/InfoListItem/InfoListItem.tsx]: can/should we expose this in classes prop for customization?

File: components/src/core/InfoListItem/InfoListItem.styles.tsx (hunk 1 above)
Comment (emclaug2) [reply components/src/core/InfoListItem/InfoListItem.styles.tsx]: This uses the theme that is passed into the MUIThemeProvider.  I tested this out in storybook by setting the InfoListItem's root bg color to use the theme primary, adjusting the light and dark themes to have a difference (light = red, dark = green), then toggling between the themes in the config.js.

File: components/src/core/InfoListItem/InfoListItem.tsx (hunk 4 above)
Comment (emclaug2) [reply components/src/core/InfoListItem/InfoListItem.tsx]: I think so, I added it in.

File: components/src/core/InfoListItem/InfoListItem.tsx (hunk 3 above)
Comment (emclaug2) [reply components/src/core/InfoListItem/InfoListItem.tsx]: Yep!

File: components/src/core/InfoListItem/InfoListItem.test.tsx (hunk 2 above)
Comment (emclaug2) [reply components/src/core/InfoListItem/InfoListItem.test.tsx]: Absolutely, I'm going to fix this test.  I'll re-request review when the tests are passing.

**Comment by emclaug2** [comment]
> I don't hate breaking out the styles into a separate file to improve readability when there's a lot of them. Not a huge fan of passing around the whole props array to different functions, but it works. You'll have to merge in your changes for the wrapping behavior since that PR went in first.

InfoListItem had about ~80 lines of logic just related to styles and more classes than most other components so I figured it would be a good candidate for a POC.  I like the idea of removing as many of the inline-styles as possible so that they don't conflict with potential style overrides that a user may be passing in.  Now a user's style overrides will always take precedence over our styles or prop inputs.

File: components/src/core/InfoListItem/InfoListItem.styles.tsx (hunk 1 above)
Comment (joebochill) [reply components/src/core/InfoListItem/InfoListItem.styles.tsx]: Would it make more sense to have a separate element that goes from transparent to this semi-transparent black color over the existing background?

E.g., assume the user has the background color set to yellow (on a dark gray page). Right now the background would change from `yellow -> almost black` versus `yellow -> dark yellow`.

Comment (emclaug2) [reply components/src/core/InfoListItem/InfoListItem.styles.tsx]: This was a good catch, I didn't test the hover state on a deep color.  I added a separate element that handles the hover event.

**Pull request merged**